api/src/search/backend/load_opportunities_to_index.py

```diff
@@ -251,6 +253,23 @@ def fetch_existing_opportunity_ids_in_index(self) -> set[int]:
             (TransportError, ConnectionTimeout)
         ),  # Retry on TransportError (including timeouts)
     )
+
+    def fetch_attachments(self, opportunity_id) -> list:
+        return self.db_session.execute(
+            select(OpportunityAttachment.file_location, OpportunityAttachment.file_name).where(
+                OpportunityAttachment.opportunity_id == opportunity_id
+            )
+        ).all()
+
+
+    def get_attachment_data(self, file_path) -> str:
+        with ExitStack() as stack:
+            file = stack.enter_context(
+                smart_open.open(file_path, "rb"),
+            )
+            file_content = file.read()
+            return  base64.b64encode(file_content).decode("utf-8")
```

Comment:
Why not just use the context manager of files themselves? ExitStack is for really complex scenarios where you need to open a lot of context managers at once. Rare that we'll ever need that.


```suggestion
        with smart_open.open(file_path, "rb") as file:
            file_content = file.read()
            return  base64.b64encode(file_content).decode("utf-8")
```